sky/skylet/job_lib.py

```diff
@@ -670,9 +668,10 @@ def tail_logs(cls,
             'else job_lib.get_latest_job_id()',
             'run_timestamp = job_lib.get_run_timestamp(job_id)',
             (f'log_dir = os.path.join({constants.SKY_LOGS_DIRECTORY!r}, '
-             'run_timestamp)'),
+             '"" if run_timestamp is None else run_timestamp)'),
             (f'log_lib.tail_logs({job_owner!r},'
-             f'job_id, log_dir, {spot_job_id!r}, follow={follow})'),
+             f'job_id, None if run_timestamp is None else log_dir, '
+             f'{spot_job_id!r}, follow={follow})'),
```

Comment:
nit: this logic looks good to me, but it seems the inline commands can cause readability problem. Should we consider changing the code generation to generate the multi-line python commands with https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Here-Documents?
```
python - <<EOF
import os
from sky.skylet import job_lib, log_lib

# generated code for the functionalities
...
EOF
```
Reference: https://stackoverflow.com/questions/2043453/executing-multi-line-statements-in-the-one-line-command-line

Reply:
Another way would be wrap the logics in a function and directly call the function in the generated code, instead of having the logic written as strings.

Reply:
Thanks. 

1. Did not get EOF to work. Many variations of 
```
        return f"""python3 - <<EOF\n{code}\nEOF"""
```
with `code` being `\n`-joined returns
```
bash: warning: here-document at line 0 delimited by end-of-file (wanted `EOF')
bash: syntax error: unexpected end of file
logout
```

2. We could do this, but I slightly worry about the skylet on an existing cluster being outdated and users failed to run `sky logs` after upgrading.

Reply:
How does it look now? Same approach, reformatted to be more readable.

Reply:
It looks much better to me! Thanks for reformating it!